src/test/ui/associated-types/associated-types-unconstrained.stderr

```diff
@@ -1,8 +1,10 @@
-error[E0284]: type annotations needed: cannot resolve `<_ as Foo>::A == _`
+error[E0284]: type annotations needed
   --> $DIR/associated-types-unconstrained.rs:14:20
    |
 LL |     let x: isize = Foo::bar();
-   |                    ^^^^^^^^
+   |                    ^^^^^^^^ cannot infer type
+   |
+   = note: cannot resolve `<_ as Foo>::A == _`
```

Comment:
[This should suggest `<isize as Foo>::bar()`](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=6d1e97dffeff3e78f2a7fd36198a8066).

Reply:
On what basis? In that playground snippet, it's the only impl, but that doesn't seem like a *common case*, does it?

Reply:
I think it should suggest all the cases that would fit, by looking at all the types that implement `Foo`, and filtering by the associated type `A == isize`. I can see that being a small enough set that we can provide structured suggestions for them.